### DISCUSSION TITLE
Hall effect sensors

### DISCUSSION THREAD

**Discussion by DG-PROPULSION** [Jehanzeb1973/Ardu_ECU#6] [Ideas]
Hi @Jehanzeb1973 
Back with another discussion. I am sort of working on possible hall effect sensors that can be used with our jet engines.
The hall effect sensors that come with the rc jet engines usually should produce a square wave. right?
BUT you noticed that some of them didn't produce a good enough square wave that could be registered as a logic "YES" and "NO"?

Now most of these rpm sensor chips (like the one from xicoy) already come with an Op-amp like LM358 installed on it which should take the signal from hall effect and amplify it to a proper square wave 0-5v.
Is that right assumption?

BUT since they are not producing a reliable 0-5v square output what are the steps we should do?

Thanks.

**Comment by DG-PROPULSION** [comment]
![hall effect sensor rc turbine](https://user-images.githubusercontent.com/128423191/229619423-28a0d1e0-4f54-4060-af43-8b7d9d9479cf.jpg)

I am talking about these hall effect rpm sensors that usually come with rc turbines and can also be bought online from xicoy etc.
Now its visible but the chip below the hall effect sensor in this picture is most probably a LM358 Op-amp.

**Reply by DG-PROPULSION** [reply]
Now should we use another LM358 Op-amp based Schmitt trigger to condition the signal coming out of this chip to get a proper 0-5v square wave?

Thanks
dstromeng

**Comment by Jehanzeb1973** [comment]
Yes, for the sensors I have, it is advisable to use another op-amp to condition the signal. 

Jehanzeb Khan
DIGIPAK Pvt. Limited
Mob: +92(0333) 5277770


http://DigiPak.org/

**Reply by DG-PROPULSION** [reply]
Hi @Jehanzeb1973 
I checked the voltage coming out of the rom sensors coming out of my rc jet engine.
It’s also somewhere around 2.5v. It’s also using a Xicoy ECU.
So I am pretty sure it’s also using the same kind of rpm sensors as yours.

Do you know if the magnets in the compressor nut are diametrically magnetised or axially magnetised?

Thanks

**Reply by DG-PROPULSION** [reply]
Here is the wave form from the sensor. Is this what you saw?
![E32EEA5D-D740-43FB-BB2E-072E86798757](https://user-images.githubusercontent.com/128423191/230690151-ebb2a372-048e-4c28-a3a3-e4515149431a.jpeg)

**Comment by Jehanzeb1973** [comment]
Hi @dstromeng
I checked a few nuts and all  are axially magnetized so you have to place the magnet on its side inside the compressor nut.
Regards

Jehanzeb Khan
DIGIPAK Pvt. Limited


http://DigiPak.org/

**Reply by DG-PROPULSION** [reply]
Hi @Jehanzeb1973 
The make ring magnets with diametric magnetization as well.
which direction magnetization are you talking of 1 or 2-
1

![1](https://user-images.githubusercontent.com/128423191/230779874-4121bb4a-fe10-4e0f-9746-956b3538bb26.png)

2

![2](https://user-images.githubusercontent.com/128423191/230779889-ec800cfb-e590-442f-887d-9b05d6e3304e.png)

Thanks.

**Comment by Jehanzeb1973** [comment]
Hi @dstromeng
Yes, I got the same results. You can see your signal Peak to Peak is 2.5 volts and centered at 2.5 volts which means your signal level is 0-5 v. As this is high for ESP32, you need to use a resistor and Reversed Biased 3.3 v Zener similar to RC input signal. You signal looks clean so you probably don’t need another opamp stage and a resistor Zener is enough for input to ESP32

Jehanzeb Khan
DIGIPAK Pvt. Limited


http://DigiPak.org/

**Reply by DG-PROPULSION** [reply]
Hi @Jehanzeb1973 
This signal is from RPM sensors not the servo.
Do you think this is a good enough digital signal as the RPM input to ESP32 or we need a LM358 based Schmitt trigger to convert it into a digital pulse?

**Comment by Jehanzeb1973** [comment]
I am talking about No 2. All the compressor nuts I have have this type of magnet

Jehanzeb Khan
DIGIPAK Pvt. Limited


http://DigiPak.org/

**Reply by DG-PROPULSION** [reply]
Hi @Jehanzeb1973 
You may need to recheck. I am pretty sure the ring magnet in the compressor nut is diametrically magnetised.

If what you are saying is true and the magnets are axially magnetised as in no. 2 picture then it cannot produce a changing magnet field which the sensor can sense.

Regards.

**Comment by Jehanzeb1973** [comment]
The signal looks good and you might be ok without LM358. However you will need 3.3V Zener and resistor to bring it to within 3.3V as this signal will reach 3.75v which is outside ESP32 limit.
REgards

Jehanzeb Khan
DIGIPAK Pvt. Limited



http://DigiPak.org/

**Reply by DG-PROPULSION** [reply]
I checked by connecting the sensor directly to the ECU running your code. It was showing erratic readings almost double of what the actual tpm was.

i then added a LM358 to the circuit and it helped. The sensor is working fine till 10000 rpm (that’s highest the starter motor can go) and both the ESP32 ECU and the Xicoy ECU were reading exactly same.

Not the actual engine test run will show if it works in  the full rpm range.
